EIPS/eip-7873.md

```diff
@@ -155,21 +155,41 @@ At the start of the block in which [EIP-3540](./eip-3540.md) activates, set the
 
 ```solidity
 {
-    /// Takes [tx_initcode_hash][salt][init_data] as input,
-    /// creates contract and returns the address or failure otherwise
+    /// Takes [tx_initcode_hash][salt][unsafe_flag][init_data] as input,
+    /// creates contract and returns the address or failure otherwise.
 
-    // init_data.length can be 0, but the first 2 words are mandatory
-    let size := calldatasize()
-    if lt(size, 64) { revert(0, 0) }
+    /// tx_initcode_hash and salt are 32 bytes wide, unsafe_flag is 1 byte wide, init_data can be any width.
+    /// If unsafe_flag is 0x00, `CALLER` is included in the salt and impacts the target address of the created account.
+    /// If unsafe_flag is non-zero, `CALLER` is excluded. Deployment by any sender will result in same target address. 
+    /// Deployments using non-zero unsafe_flag are susceptible to front-running.
 
-    calldatacopy(0, 0, size)  // copy tx_initcode_hash, salt and init_data to memory to hash
-    let final_salt := keccak256(0, size)
+    // init_data.length can be 0, but the first 65 bytes are mandatory
+    let size := calldatasize()
+    if lt(size, 65) { revert(0, 0) }
+
+    // copy tx_initcode_hash, salt, unsafe_flag and init_data to memory to hash.
+    // note that the unsafe_flag is included to prevent hash collisions between the two branches:
+    //   - unsafe_flag == 0, caller's address is appended by the code
+    //   - unsafe_flag == 1, caller's address is sent via init_data
```

Comment:
This is a little weird. Nothing forces the caller to include their address in `init_data`. In most cases that use non-zero unsafe flag the deployment should be totally independent of the caller. Additionally, the flag can be any non-zero value.

Reply:
I think this comment means that without hashing `unsafe_flag` the deployment with `unsafe_flag == 0` collides with deployment with `unsafe_flag == 1` and caller's address in `init_data`

Reply:
Ok, that makes more sense.